### PR TITLE
Add user api endpoints

### PR DESCRIPTION
**Pull request by JustinKnueppel**

### CODE REVIEW

File: pages/api/user.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    const userWithEmail = await prisma.user.findFirst({
+      where: { email: email },
+    });
+    if (userWithEmail) {
+      return res.status(409).end();
+    }
+
+    const user = await prisma.user.create({
+      data: { fName: firstName, lname: lastName, email: email },
```
Comment (seanmesserly) [inline pages/api/user.ts:29]: capitalization inconsistencies between `fName` and `lname`

File: pages/api/users/[userId].ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "../../../lib/prisma";
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { userId: userIdParam } = req.query;
+  if (userIdParam instanceof Array) {
```
Comment (seanmesserly) [inline pages/api/users/[userId].ts:9]: Out of curiosity, was it parsing as an array somehow? I would have guessed they had to be strings

File: pages/api/users/[userId].ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    return res.status(404).end();
+  }
+  const userId = parseInt(userIdParam);
+  if (isNaN(userId)) {
+    return res.status(404).end();
+  }
+
+  if (req.method === "GET") {
+    const user = await prisma.user.findUnique({ where: { id: userId } });
+    return res.status(200).json(user);
```
Comment (seanmesserly) [inline pages/api/users/[userId].ts:19]: Should this status code be dependent on whether a user was actually found or not?

**Review by seanmesserly — CHANGES_REQUESTED** [review]

File: pages/api/users/[userId].ts (hunk 2 above)
Comment (JustinKnueppel) [reply pages/api/users/[userId].ts:9]: The type in typescript was string or array of strings, not sure why

File: pages/api/users/[userId].ts (hunk 3 above)
Comment (JustinKnueppel) [reply pages/api/users/[userId].ts]: Added

File: pages/api/user.ts (hunk 1 above)
Comment (JustinKnueppel) [reply pages/api/user.ts]: Fixed

File: pages/api/users/[userId].ts (hunk 2 above)
Comment (JustinKnueppel) [reply pages/api/users/[userId].ts:9]: I think this is because it can include params passed at the end of the uri via foo=bar and there can be multiple

**Review by seanmesserly — APPROVED** [review]

**Pull request merged**